runtime/compiler/trj9/env/PersistentCHTable.cpp

```diff
@@ -301,30 +304,46 @@ TR_PersistentCHTable::findClassInfo(TR_OpaqueClassBlock * classId)
    return cl;
    }
 
-
+/**
+ * Find persistent JIT class information for a given class.
+ * The class table lock is used to synchronize use of this method
+ */
 TR_PersistentClassInfo *
 TR_PersistentCHTable::findClassInfoAfterLocking(
       TR_OpaqueClassBlock *classId,
       TR::Compilation *comp,
       bool returnClassInfoForAOT)
    {
-   if (comp->fej9()->isAOT_DEPRECATED_DO_NOT_USE() && !returnClassInfoForAOT) // for AOT do not use the class hierarchy
+   return findClassInfoAfterLocking(classId, comp->fe(), returnClassInfoForAOT);
+   }
+
+/**
+ * Find persistent JIT class information for a given class.
+ * The class table lock is used to synchronize use of this method
+ */
+TR_PersistentClassInfo *
+TR_PersistentCHTable::findClassInfoAfterLocking(
+      TR_OpaqueClassBlock *classId,
+      TR_FrontEnd *fe,
+      bool returnClassInfoForAOT)
+   {
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(fe);
+   if (fej9->isAOT_DEPRECATED_DO_NOT_USE() && !returnClassInfoForAOT) // for AOT do not use the class hierarchy
       return NULL;
 
-   if (comp->getOption(TR_DisableCHOpts))
+   if (TR::comp()->getOption(TR_DisableCHOpts))
```

Comment:
If this code is executed on a non-compilation thread (or when comp object does not yet exist) it will crash